tcms/rpc/tests/test_testcase.py

```diff
@@ -522,16 +620,47 @@ def test_add_component_from_another_product_is_not_allowed(self):
             self.rpc_client.TestCase.add_component(self.test_case.pk, self.bad_component.name)
 
 
-class TestCaseAddComment(APITestCase):
+class TestRemoveComponentPermission(APIPermissionsTestCase):
+    permission_label = 'testcases.delete_testcasecomponent'
+
+    def _fixture_setup(self):
+        super()._fixture_setup()
+        self.test_case = TestCaseFactory()
+        self.good_component = ComponentFactory(product=self.test_case.category.product)
+        self.bad_component = ComponentFactory(product=self.test_case.category.product)
+        self.test_case.add_component(self.good_component)
+        self.test_case.add_component(self.bad_component)
+
+    def verify_api_with_permission(self):
+        self.rpc_client.TestCase.remove_component(self.test_case.pk,
+                                                  self.bad_component.pk)
+        result = self.test_case.component.all()
+        self.assertEqual(len(result), 1)
```

Comment:
result.count() is better (and faster) b/c len() will evaluate the entire queryset and then count how many objects are there while .count() will do the counting directly in the database.

Reply:
As far as I know, if we use count, then two queries to the database will be made - one for `count` and another for `result.first()`. Using `len` will evaluate the query and `result.first()` will be served from the cache.

Please let me know if I have the wrong understanding of the Queryset API I will make the requested changes. But, I think current implementation is quicker.

Reply:
Counting in DB is always quicker than counting in Python plus we like readability. Given `result` is of type queryset then use it as it was intended to be used.